### PR TITLE
the Cache

### PR DESCRIPTION
**Pull request by bvavala**

### CODE REVIEW

**Comment by bvavala** [comment]
- agreed for the PR, but commit descr actually summarize the result
- kind of right, though it frees up items only when it's full and others must be loaded
- yes I do but... the code must be refactored due to the dependicies among the components. Given the amount of code that has to be touched, this can be left for a separate PR

File: common/state/state_kv.cpp [hunk 1]
```diff
@@ -604,12 +575,13 @@ void pstate::trie_node::do_read_value(
...
 {
-    pdo::error::ThrowIf<pdo::error::RuntimeError>(
-        !header->hasChild, "delete value, header must have child");
-    block_offset current_child_bo;
-    current_child_bo.deserialize_offset(*goto_child_offset(header));
+    block_offset current_child_bo(node.node.child_offset);
+    //current_child_bo.deserialize_offset(*goto_child_offset(header));
+
+    //TODO MODIFY DELETE TO WRITE A DELETED VALUE HEADER (1 byte)
+    //TODO remove bytearray from from delete value in data node
```
Comment (cmickeyb) [inline common/state/state_kv.cpp]: remove all TODOs before you commit. these can stay in your own repo, but don't commit them.

Comment (cmickeyb) [reply common/state/state_kv.cpp]: You are going to make me crazy.

**Comment by cmickeyb** [comment]
How do I test that the cache is working? Did you modify and/or add tests that would verify this?

**Comment by bvavala** [comment]
added most-recently-used cache replacement algorithm for better performance using tests in #143

**Comment by cmickeyb** [comment]
why do we need three different policies compiled into the code when only one is going to be called?

**Comment by bvavala** [comment]
updates:
- 1 policy
- cache tests enabled
- tests dependent on cache and block size

**Pull request merged**